vector/src/main/java/im/vector/app/features/home/room/list/home/HomeRoomListFragment.kt

```diff
@@ -0,0 +1,226 @@
+                roomListViewModel.handle(HomeRoomListAction.ChangeRoomNotificationState(quickAction.roomId, RoomNotificationState.MUTE))
+            }
+            is RoomListQuickActionsSharedAction.Settings -> {
+                navigator.openRoomProfile(requireActivity(), quickAction.roomId)
+            }
+            is RoomListQuickActionsSharedAction.Favorite -> {
+                roomListViewModel.handle(HomeRoomListAction.ToggleTag(quickAction.roomId, RoomTag.ROOM_TAG_FAVOURITE))
+            }
+            is RoomListQuickActionsSharedAction.LowPriority -> {
+                roomListViewModel.handle(HomeRoomListAction.ToggleTag(quickAction.roomId, RoomTag.ROOM_TAG_LOW_PRIORITY))
+            }
+            is RoomListQuickActionsSharedAction.Leave -> {
+                roomListViewModel.handle(HomeRoomListAction.LeaveRoom(quickAction.roomId))
+                promptLeaveRoom(quickAction.roomId)
+            }
+        }
+    }
+
+    private fun promptLeaveRoom(roomId: String) {
+        val isPublicRoom = roomListViewModel.isPublicRoom(roomId)
+        val message = buildString {
+            append(getString(R.string.room_participants_leave_prompt_msg))
+            if (!isPublicRoom) {
+                append("\n\n")
+                append(getString(R.string.room_participants_leave_private_warning))
+            }
+        }
+        MaterialAlertDialogBuilder(requireContext(), if (isPublicRoom) 0 else R.style.ThemeOverlay_Vector_MaterialAlertDialog_Destructive)
+                .setTitle(R.string.room_participants_leave_prompt_title)
+                .setMessage(message)
+                .setPositiveButton(R.string.action_leave) { _, _ ->
+                    roomListViewModel.handle(HomeRoomListAction.LeaveRoom(roomId))
+                }
+                .setNegativeButton(R.string.action_cancel, null)
+                .show()
+    }
+
+    private fun getAdapterForData(data: HomeRoomSection): EpoxyControllerAdapter {
+        return when (data) {
+            is HomeRoomSection.RoomSummaryData -> {
+                RoomSummaryPagedController(
+                        roomSummaryItemFactory,
+                        RoomListDisplayMode.ROOMS
+                ).also { controller ->
+                    controller.listener = this
+                    data.list.observe(viewLifecycleOwner) { list ->
+                        controller.submitList(list)
+                    }
+                }.adapter
+            }
+        }
+    }
+
+    private fun handleSelectRoom(event: HomeRoomListViewEvents.SelectRoom, isInviteAlreadyAccepted: Boolean) {
+        navigator.openRoom(
+                context = requireActivity(),
+                roomId = event.roomSummary.roomId,
+                isInviteAlreadyAccepted = isInviteAlreadyAccepted,
+                trigger = ViewRoom.Trigger.RoomList
+        )
+    }
+
+    // region RoomListListener
+
+    override fun onRoomClicked(room: RoomSummary) {
```

Comment:
Not to be actioned, but I always wondered how feasible it is to lift interaction events like these ones into a separate class. What are your thoughts?

Reply:
Was thinking about this also, but for now I decided to keep it as it is. Will take a look when everything else will be complete